### PR TITLE
Improve docs for setting up a Gatsby application that pulls content from Markdown files

### PR DESCRIPTION
**Pull request by paschalidi**
👋 hello you all, hope you are all having a great day! 🌳 🌸 

## Description
As discussed here https://github.com/gatsbyjs/gatsby/issues/13710 the documentation for creating a Gatsby application that will render content from Markdown files has some space for improvement.

This PR tries to make the tutorial easier to follow by giving more specific steps to the user and by adding an example with name `using-markdown-files`. This way the users can see how what is being said in the tutorial fits in a real Gatsby application. 

In https://github.com/gatsbyjs/gatsby/issues/13710 there are five "Recommendations". From those five, this PR tackles four. To be specific:

1. Install gatsby-source-filesystem: Configuring but no suggested path to mkdir at (e.g. /blog/). Add more explicit guidance on this page.
2. Note on creating Markdown files: This is not very helpful in terms of where to place the Markdown files (in /articles?). Add more explicit guidance on YYYY-MM-DD-lorem-ipsum-dolor-sit-amet/index.md file naming.
3. Tone: Sideline some of the extraneous information that we don’t need as a note, e.g. "Gatsby calls the createPages API (if present)".
4. Style: Correct minor issues (punctuation is inconsistent; periods at ends of headings and hyphens instead of em-dashes).  

I would like to address the  
* General: Add visibility by retitling to "how to create a blog" or some such title instead of "Querying your data with GraphQL".

in a different PR if there are no objection into that. 

## Related Issues
Related https://github.com/gatsbyjs/gatsby/issues/13710

I am happy receiving any constructing feedback from you! 
Big thanks to the people that took the time to report and organise that! 🥇

### CODE REVIEW

**Comment by freiksenet** [comment]
CC @prestonso.

**Comment by paschalidi** [comment]
@marcysutton thanks a lot for taking the time to review this! Very nice suggestions onto the point! Will stay tuned ofc! I will finish with the rest in the weekend.

**Comment by paschalidi** [comment]
@marcysutton again thanks for the amazing review! Please have another look when you get the time :)

File: docs/docs/adding-markdown-pages.md [hunk 1]
```diff
@@ -9,41 +9,51 @@ Here are the steps Gatsby follows for making this happen.
...
 1.  Read files into Gatsby from the filesystem
 2.  Transform markdown to HTML and frontmatter to data
-3.  Create a page component for the markdown files
-4.  Programmatically create pages using Gatsby's Node.js `createPage` API
+3.  Add a markdown file
+4.  Create a page component for the markdown files
+5.  Create static pages using Gatsby's Node.js `createPage` API
 
-### Read files into Gatsby from the filesystem - `gatsby-source-filesystem`
+### Read files into Gatsby from the filesystem — `gatsby-source-filesystem`
```
[the remark this answers is not in the document]
Comment (marcysutton) [reply docs/docs/adding-markdown-pages.md]: Hi there, we need to update the headings in these docs to use the correct hierarchy: they should start with h2, and go in order without skipping levels. This ensures the page can be easily digested by users of assistive technology: https://webaim.org/techniques/semanticstructure/

**Comment by marcysutton** [comment]
I pushed some changes to make this align with the Gatsby Style Guide. I think it's ready to go now! 

I agree that retitling or recategorizing it for visibility would be helpful–it is not at all intuitive to look for this in the GraphQL section. But that can come in a separate PR!

**Comment by paschalidi** [comment]
Hey Marcy thanks for taking such good care here and thanks for caring for quality! 🙇‍♀

**Comment by marcysutton** [comment]
You're welcome, @paschalidi! Thanks for your wonderful contribution!

**Pull request merged**